src/components/text_input/text_input.ts

```diff
@@ -56,17 +57,30 @@ export class TextInput extends Component<Props, SpreadsheetChildEnv> {
           this.save();
         }
       },
-      {
-        capture: true,
-      }
+      { capture: true }
     );
   }
 
   onChange() {
     this.save();
   }
 
+  onFocus() {
+    this.focusInputAndSelectContent();
+  }
+
+  onBlur() {
+    this.save();
+  }
+
   save() {
     this.props.onChange((this.inputRef.el?.value || "").trim());
   }
+
+  private focusInputAndSelectContent() {
```

Comment:
This seems to only work half of the times somehow

https://drive.google.com/file/d/1OoW4AUPQMZ16YazZCCD694OlNYUqim4Z/view?usp=drive_link

Reply:
Hello, Sorry for the delay. I spent some time investigating the issue and working on other tasks.

I have identified that the root cause is related to the `mouseup` event. [When text is selected, the mouseup event causes the text to be deselected.](https://stackoverflow.com/questions/23407067/select-not-functioning-correctly-in-chrome)

Can we use `t-on-mouseup.prevent` to address this issue? Or, do you have any other solutions you have encountered in the past that could resolve this?

Reply:
In the `bottom_bar_sheet` `focusInputAndSelectContent` we change the selection manually isntead of using `.select()`. Maybe that'd work ? Otherwise yeah I guess a preventDefault is fine. I'd still add a comment somewhere explaining why we have to do this

Reply:
I’ve tried this as well, but since the `mouseup` event fires immediately after `focus`, it still causes an issue. I’ve added the comment, and the code is now ready for re-validation :)